hack/olm-registry/hypershift-template.yaml

```diff
@@ -0,0 +1,222 @@
+                              olm.operatorframework.io/exclude-global-namespace-resolution: "true"
+                          spec:
+                            upgradeStrategy: TechPreviewUnsafeFailForward
+                            targetNamespaces:
+                              - openshift-deployment-validation-operator
+                      - complianceType: MustHave
+                        objectDefinition:
+                          apiVersion: operators.coreos.com/v1alpha1
+                          kind: Subscription
+                          metadata:
+                            name: deployment-validation-operator
+                            namespace: openshift-deployment-validation-operator
+                          spec:
+                            channel: ${CHANNEL}
+                            config:
+                              affinity:
+                                nodeAffinity: {}
+                              env:
+                              - name: "NAMESPACE_IGNORE_PATTERN"
+                                value: "^(openshift.*|kube-.*|default|dedicated-admin|redhat-.*|acm|addon-dba-operator|codeready-.*|prow)$"
+                            name: deployment-validation-operator
+                            source: deployment-validation-operator-catalog
+                            sourceNamespace: openshift-deployment-validation-operator
+                      - complianceType: MustHave
+                        objectDefinition:
+                          apiVersion: networking.k8s.io/v1
+                          kind: NetworkPolicy
+                          metadata:
+                            name: allow-from-openshift-insights
+                            namespace: openshift-deployment-validation-operator
+                          spec:
+                            podSelector: {}
+                            ingress:
+                              - from:
+                                - namespaceSelector:
+                                    matchLabels:
+                                      name: openshift-insights
+                            policyTypes:
+                              - Ingress
+                      - complianceType: MustHave
+                        objectDefinition:
+                          apiVersion: networking.k8s.io/v1
+                          kind: NetworkPolicy
+                          metadata:
+                            name: allow-from-openshift-olm
+                            namespace: openshift-deployment-validation-operator
+                          spec:
+                            podSelector: {}
+                            ingress:
+                              - from:
+                                - namespaceSelector:
+                                    matchLabels:
+                                      name: openshift-operator-lifecycle-manager
+                            policyTypes:
+                              - Ingress
```

Comment:
Spec to be changed to:
```
                          spec:
                            podSelector:
                              matchLabels:
                                olm.catalogSource: deployment-validation-operator-catalog
                            ingress:
                              - {}
                            policyTypes:
                              - Ingress
```
OLM pod is indeed running in the HCP namespace (so not in the `openshift-operator-lifecycle-manager` ns) on the MC cluster and is not visible in the dataplane.

Solution here is:
- To allow all external traffic.
- But only to the catalog pod.

Reply:
Update thank you